src/player.cpp

```diff
@@ -284,12 +286,18 @@ player::player() : Character()
     reset_encumbrance();
 
     morale.reset( new player_morale() );
+    last_craft.reset( new craft_command() );
 }
 
 player::~player()
 {
 }
 
+player::player(const player &) = default;
+player::player(player &&) = default;
+player &player::operator=(const player &) = default;
+player &player::operator=(player &&) = default;
```

Comment:
This may allow replacing `player_morale_ptr` with a `copyable_unique_ptr`. It was implemented as separate class so its constructor/destructor (requiring access to the moral class) could be in the cpp file.